src/auth/maptiler_hmacsha256/core/qgsauthmaptilerhmacsha256method.cpp

```diff
@@ -70,26 +71,38 @@ bool QgsAuthHmacSha256Method::updateNetworkRequest( QNetworkRequest &request, co
     return false;
   }
 
-  const QString key = mconfig.config( QStringLiteral( "key" ) );
   const QString token = mconfig.config( QStringLiteral( "token" ) );
+  const QStringList splitToken = token.split( '_' );
 
-  QString baseUrl = request.url().toString();
+  if ( splitToken.count() != 2 )
+  {
+    QgsDebugMsg( QStringLiteral( "Update request config FAILED for authcfg: %1: config invalid" ).arg( authcfg ) );
+    return false;
+  }
+
+  const QString key = splitToken.at( 0 );
+  const QString secret = splitToken.at( 1 );
 
-  QString keyedUrl = baseUrl + QStringLiteral( "?key=" ) + key;
+  QUrl url = request.url();
+  QUrlQuery query( url.query() );
+  QList<QPair<QString, QString> > queryItems = query.queryItems();
+  queryItems.append( {QStringLiteral( "key" ), key} );
```

Comment:
does this work correctly if the URL template already contains "key" ? e.g. if my URL template is `https://..../{z}/{x}/{y}.png?key=blahblah` - are we going to remove `key=blahblah` or only append to it?

Reply:
oops... I've forgot this...